lib/iris/fileformats/netcdf/saver.py

```diff
@@ -2509,6 +2583,39 @@ def store(data, cf_var, fill_value):
             )
             warnings.warn(msg.format(cf_var.name, fill_value))
 
+    def _deferred_save(self):
+        """
+        Create a 'delayed' to trigger file completion for lazy saves.
+
+        This contains all the deferred writes, which complete the file by filling out
+        the data of variables initially created empty.
+
+        """
+        # Create a lock to satisfy the da.store call.
+        # We need a serialisable lock for scheduling with processes or distributed.
+        # See : https://github.com/dask/distributed/issues/780
+        # However, this does *not* imply safe access for file writing in parallel.
+        # For that, DeferredSaveWrapper uses a filelock as well.
+        lock = SerializableLock()
+
+        # Create a single delayed da.store operation to complete the file.
+        sources, targets = zip(*self.deferred_writes)
+        result = da.store(sources, targets, compute=False, lock=lock)
```

Comment:
Some of the sources can be numpy array here, so one could do something like:
```suggestion
        result = da.store([da.asarray(s) for s in sources], targets, compute=False, lock=lock)
```

Reply:
See [comment above](https://github.com/SciTools/iris/pull/5031#issuecomment-1322340895)
I think this is the same issue ?

Reply:
Yes, it's also related to having real data. One needs to make sure these are converted to Dask arrays.